python/ray/train/v2/tests/test_metrics.py

```diff
@@ -9,43 +9,120 @@
-    self._metrics.train_report_total_blocked_time_s += value
+    self._metrics_tracker.record(WorkerMetrics.REPORT_TOTAL_BLOCKED_TIME_S, value)
+
+
+def mock_time_monotonic(monkeypatch, time_values: list[float]):
+    time_index = 0
+
+    def mock_time():
+        nonlocal time_index
+        value = time_values[time_index]
+        time_index = time_index + 1
+        return value
+
+    monkeypatch.setattr(
+        ray.train.v2._internal.callbacks.metrics, "time_monotonic", mock_time
+    )
+
+
+def test_time_metric(monkeypatch):
+    """Test TimeMetric functionality."""
+    monkeypatch.setattr(ray.train.v2._internal.metrics.base, "Gauge", MockGauge)
+
+    base_tags = {"run_name": "test_run"}
+    metric = TimeMetric(
+        name="test_time",
+        description="Test time metric",
+        base_tags=base_tags,
+    )
+
+    # Test recording values
+    metric.record(1.0)
+    assert metric.get_value() == 1.0
+
+    # Test updating metric
+    metric.record(2.0)
+    assert metric.get_value() == 3.0
+
+    # Test reset
+    metric.reset()
+    assert metric.get_value() == 0.0
 
 
-def mock_on_worker_group_event(self, value: float, event: str):
-    """Mock function to set the value of the worker group event"""
-    if event == "start":
-        self._metrics.train_worker_group_start_total_time_s += value
-    elif event == "shutdown":
-        self._metrics.train_worker_group_shutdown_total_time_s += value
+def test_enum_metric(monkeypatch):
+    """Test EnumMetric functionality."""
+    monkeypatch.setattr(ray.train.v2._internal.metrics.base, "Gauge", MockGauge)
+
+    class TestEnum(enum.Enum):
+        A = "A"
+        B = "B"
+        C = "C"
+
+    base_tags = {"run_name": "test_run"}
+    metric = EnumMetric[TestEnum](
+        name="test_enum",
+        description="Test enum metric",
+        base_tags=base_tags,
+        enum_tag_key="state",
+    )
+
+    # Test recording values
+    metric.record(TestEnum.A)
+    assert metric.get_value(TestEnum.A) == 1
+    assert metric.get_value(TestEnum.B) == 0
+    assert metric.get_value(TestEnum.C) == 0
+
+    metric.record(TestEnum.B)
+    assert metric.get_value(TestEnum.A) == 0
+    assert metric.get_value(TestEnum.B) == 1
+    assert metric.get_value(TestEnum.C) == 0
+
+    metric.record(TestEnum.C)
+    assert metric.get_value(TestEnum.A) == 0
+    assert metric.get_value(TestEnum.B) == 0
+    assert metric.get_value(TestEnum.C) == 1
+
+    # Test reset
+    metric.reset()
+    assert metric.get_value(TestEnum.A) == 0
+    assert metric.get_value(TestEnum.B) == 0
+    assert metric.get_value(TestEnum.C) == 0
 
 
 def test_worker_metrics_callback(monkeypatch):
-    monkeypatch.setattr(WorkerMetricsCallback, "LOCAL_METRICS_PUSH_INTERVAL_S", 0.05)
-    monkeypatch.setattr(WorkerMetricsCallback, "on_report", mock_on_report)
+    monkeypatch.setattr(ray.train.v2._internal.metrics.base, "Gauge", MockGauge)
+    mock_time_monotonic(monkeypatch, [0.0, 1.0, 10.0, 12.0])
```

Comment:
so is this a total of 3 seconds? (0-1, 10-12). Can you add a comment about this, or make it more clear by having the list be tuples of start,end?